File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        map<int, int> freqmap;
        int maxc = 0;
        for (auto& i : nums) {
            freqmap[i]++;
            maxc = max(freqmap[i], maxc);
        }
        vector<vector<int>> mp(maxc + 1);
        for (auto i : freqmap) {
            mp[i.second].push_back(i.first);
        }
        vector<int> ans;
        for (int i = maxc; i > 0 && ans.size() < k; --i) {
            for (int num : mp[i]) {
                ans.push_back(num);
                if (ans.size() == k)
                    break;
            }
        }

        return ans;
    }
};

File: Data Structures & Algorithms/first-missing-positive/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    firstMissingPositive(nums) {
        for (let i = 0; i < nums.length; i++) {
            if (nums[i] < 1 || nums[i] > nums.length) { nums[i] = -1; continue; }
            if (nums[i] == i + 1) continue;
            let temp=nums[i];
            nums[i]=nums[temp-1];
            nums[temp-1]=temp;
            if(temp!=nums[i])
            i--;
        }

        let ans = 1;
        for (let n of nums) {
            if (ans === n) {
                ans++;
            }
        }
        return ans;
    }
}
